### ISSUE TITLE
🔨 New `tag` Gradle task

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#365]
## 📝 Description

We would like to create a Gradle task named `tag` for creating a [Git annotated tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) from the project's version.

Also, this task should replace the [`gitTag.sh`][scripts/gitTag] script.

For better code reuse, this task should be added in a new `kotools.types.git` Gradle plugin and should be applied on the root project.

## 🔗 Dependencies

This issue is blocked by the following items:

- [x] #580 

## ✅ Checklist

- [ ] Create the `kotools.types.git` Gradle plugin.
- [ ] Create the `tag` Gradle task.
- [ ] Update releases issues and the dedicated template for using the `tag` Gradle task instead of the [`gitTag.sh`][scripts/gitTag] script.
- [ ] Remove the [`gitTag.sh`][scripts/gitTag] scripts.

[scripts/gitTag]: https://github.com/kotools/types/blob/main/scripts/gitTag.sh

**Comment by LVMVRQUXL** [comment]
Already done [here](https://github.com/kotools/types/blob/5a9a59f5a8b94a20bb016956c2c83687cb0d1d8b/build.gradle.kts#L25-L32).
